ParkingLotApi/Controllers/ParkingLotController.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using ParkingLotApi.Dtos;
+using ParkingLotApi.Repository;
+using ParkingLotApi.Services;
+
+namespace ParkingLotApi.Controllers
+{
+    [ApiController]
+    [Route("parkingLots")]
+    public class ParkingLotController : ControllerBase
+    {
+        private readonly ParkingLotService parkingLotService;
+
+        public ParkingLotController(ParkingLotService parkingLotService)
+        {
+            this.parkingLotService = parkingLotService;
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<ParkingLotDto>> GetById(int id)
+        {
+            var parkingLotDto = await this.parkingLotService.GetById(id);
+            return Ok(parkingLotDto);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<ParkingLotDto>>> GetALL(int?pageSize, int?startPage)
+        {
+            if (pageSize.HasValue && startPage.HasValue)
+            {
+                var parkingLotDtos = await this.parkingLotService.GetOnPage(pageSize.Value, startPage.Value);
+                return Ok(parkingLotDtos);
+            }
+
+            var parkingLotDtosAll = await this.parkingLotService.GetAll();
+            return Ok(parkingLotDtosAll);
+        }
```

Comment:
When pagesize not provided, I should still get information at pageIndex of size 15